pkg/ibm_plugin/server/plugin_manual_test.go

```diff
@@ -0,0 +1,300 @@
+	},
+	// ssh to accept ssh connection
+	{
+		Name:      "inboundSSH",
+		Direction: paragliderpb.Direction_INBOUND,
+		SrcPort:   22,
+		DstPort:   22,
+		Protocol:  6,
+		Targets:   []string{"0.0.0.0/0"},
+	},
+	//All protocol to allow all egress traffic
+	{
+		Name:      "outboundALL",
+		Direction: paragliderpb.Direction_OUTBOUND,
+		SrcPort:   -1,
+		DstPort:   -1,
+		Protocol:  -1,
+		Targets:   []string{"0.0.0.0/0"},
+	},
+}
+
+// go test --tags=ibm -run TestCreateNewResource
+func TestCreateNewResource(t *testing.T) {
+	// Notes for tester:
+	// to change region set the values below according to constants above, e.g.:
+	// - test arguments for EU-DE-1:
+	// image, zone, instanceName, resourceID := testImageEUDE, testZoneEUDE1, testInstanceNameEUDE1, testResourceIDEUDE1
+	// - test arguments for us-east-2:
+	// image, zone, instanceName, resourceID := testImageUSEast, testZoneUSEast2, testInstanceNameUSEast2, testResourceIDUSEast2
+	// - test arguments for us-south-1:
+	// image, zone, instanceName, resourceID := testImageUSSouth, testZoneUSSouth1, testInstanceNameUSSouth1, testResourceIDUSSouth1
+	// image, zone, instanceName, resourceID := testImageUSEast, testZoneUSEast1, testInstanceNameUSEast1, testResourceIDUSEast1
+	image, zone, instanceName := testImageUSEast, testZoneUSEast1, testInstanceNameUSEast1
+
+	_, fakeControllerServerAddr, err := fake.SetupFakeOrchestratorRPCServer(utils.IBM)
+	if err != nil {
+		t.Fatal(err)
+	}
+	imageIdentity := vpcv1.ImageIdentityByID{ID: &image}
+	zoneIdentity := vpcv1.ZoneIdentityByName{Name: &zone}
+	myTestProfile := string(testProfile)
+
+	testPrototype := &vpcv1.InstancePrototypeInstanceByImage{
+		Image:   &imageIdentity,
+		Zone:    &zoneIdentity,
+		Name:    core.StringPtr(instanceName),
+		Profile: &vpcv1.InstanceProfileIdentityByName{Name: &myTestProfile},
+	}
+
+	s := &IBMPluginServer{
+		orchestratorServerAddr: fakeControllerServerAddr,
+		cloudClient:            make(map[string]*sdk.CloudClient)}
+
+	description, err := json.Marshal(vpcv1.CreateInstanceOptions{InstancePrototype: vpcv1.InstancePrototypeIntf(testPrototype)})
+	require.NoError(t, err)
+
+	resource := &paragliderpb.CreateResourceRequest{Deployment: &paragliderpb.ParagliderDeployment{Id: testDeployment, Namespace: testNamespace}, Description: description}
+	resp, err := s.CreateResource(context.Background(), resource)
+	if err != nil {
+		println(err)
+	}
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+}
+
+// usage: go test --tags=ibm -run TestGetPermitList
+func TestGetPermitRules(t *testing.T) {
+	resourceID := testResourceIDUSEast1 // replace as needed with other IDs, e.g. testResourceIDEUDE1
+
+	s := &IBMPluginServer{cloudClient: make(map[string]*sdk.CloudClient)}
+
+	resp, err := s.GetPermitList(context.Background(), &paragliderpb.GetPermitListRequest{Resource: resourceID,
+		Namespace: testNamespace})
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+
+	b, err := json.MarshalIndent(resp, "", "  ")
```

Comment:
What is this checking for? And why `MarshalIndent` and not `Marshal`

Reply:
I remember switching to it due to an issue with `Marshal`. Don't remember the exact issue.